src/bundle/Resources/config/routing.yml

```diff
@@ -479,6 +479,14 @@ ibexa.rest.create_location:
 
 # Content types
 
+ibexa.rest.content_type.copy:
+    path: /content/types/{contentTypeId}
+    defaults:
+        _controller: Ibexa\Rest\Server\Controller\ContentType:copyContentType
+    condition: 'is_content_type_compatible(request, "application/vnd.ibexa.api.CopyContentTypeInput")'
```

Comment:
The feature implemented in this PR is already available within expressions in Symfony routing component:
https://symfony.com/doc/5.x/routing.html#matching-expressions

See:
```yaml
# config/routes.yaml
contact:
    path:       /contact
    controller: 'App\Controller\DefaultController::contact'
    condition:  "context.getMethod() in ['GET', 'HEAD'] and request.headers.get('User-Agent') matches '/firefox/i'"
    # expressions can also include configuration parameters:
    # condition: "request.headers.get('User-Agent') matches '%app.allowed_browsers%'"
    # expressions can even use environment variables:
    # condition: "context.getHost() == env('APP_MAIN_HOST')"
```

In the above case :point_up: `User-Agent` header was used, but you can use `Content-Type` just as well.

```suggestion
    condition: 'request.headers.get("Content-Type") matches "~application\/vnd\.ibexa\.api\.CopyContentTypeInput~",
    # in Symfony 7:
    condition: 'request.headers.get("Content-Type") starts with "application/vnd.ibexa.api.CopyContentTypeInput"'
```

Reply:
@Steveb-p we were discussing this, but we decided to go with a dedicated approach to make this condition less painful to write.

Reply:
I've sync'ed with @alongosz about it, and since that is the team decision, **I have to yield here.**

I will however note that, IMO this does not provide the simplicity we seek, and introduces a technical debt (a dedicated function code, expression function, tests) that would not be otherwise required.

Reply:
> @Steveb-p we were discussing this, but we decided to go with a dedicated approach to make this condition less painful to write.

TBH I was expecting this to be even simpler, sth like 
```yaml
condition: 'ibexa_is_media_type(request, "CopyContentTypeInput")'
```
Also I'd prefer if the name started with the "ibexa_" prefix.

@barw4  We can probably discuss this further internally since @Steveb-p is back from vacation.

Reply:
Alternatively, a getter could be created:

```yaml
condition: 'ibexa_get_media_type(request) == "CopyContentTypeInput"'
```